packages/dds/merge-tree/src/MergeTreeTextHelper.ts

```diff
@@ -0,0 +1,172 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { IIntegerRange } from "./base";
+import { ISegment, Marker } from "./mergeTreeNodes";
+import { refHasTileLabel } from "./referencePositions";
+import { MergeTree } from "./mergeTree";
+import { IMergeTreeTextHelper, TextSegment } from "./textSegment";
+
+interface ITextAccumulator {
+    textSegment: TextSegment;
+    placeholder?: string;
+    parallelArrays?: boolean;
+}
+
+interface ITextAndMarkerAccumulator extends ITextAccumulator {
+    parallelArrays: true;
+    parallelText: string[];
+    parallelMarkers: Marker[];
+    parallelMarkerLabel: string;
+    tagsInProgress: string[];
+}
+
+function isTextAndMarkerAccumulator(accum: ITextAccumulator): accum is ITextAndMarkerAccumulator {
+    return accum.parallelArrays === true;
+}
+
+type ITextAccumulatorType = ITextAccumulator | ITextAndMarkerAccumulator;
+
+/**
+ * @deprecated  for internal use only. public export will be removed.
+ * @internal
+ */
+export class MergeTreeTextHelper implements IMergeTreeTextHelper {
```

Comment:
all copy pasted from text segment. added an interface so we don't need to export this class which is coupled to merge tree